ckanext/googleanalytics/model.py

```diff
@@ -371,7 +371,240 @@ def get_all_visits(cls,id):
         }
         return results
 
+class AudienceLocation(Base):
+    """
+    Contains stats for different visitors locations
+    Stores all different countries.
+    """
+    __tablename__ = 'audience_location'
+
+    id = Column(types.Integer, primary_key=True, autoincrement=True, unique=True)
+    location_name = Column(types.UnicodeText, nullable=False, primary_key=True)
+
+    visits_by_date = relationship("AudienceLocationDate", back_populates="location")
+
+    @classmethod
+    def get(cls, id):
+        return model.Session.query(cls).filter(cls.id == id).first()
+
+    @classmethod
+    def update_location(cls, name):
+        # Check if the location can be found
+        location = model.Session.query(cls).filter(cls.location_name == location_name)
+        if location is None:
+            # Add location if not in db
+            location = AudienceLocation(location_name=location_name)
+            model.Session.add(location)
+            log.debug("New location added: %s", location_name)
+        else:
+            location.location_name = name
+            log.debug("Location name updated: %s", location_name)
+
+        model.Session.flush()
+        return True
+
+class AudienceLocationDate(Base):
+    """
+    Contains stats for different visitors locations by date
+    Maps user amounts to dates and locations
+    """
+    __tablename__ = 'audience_location_date'
+
+    id = Column(types.Integer, primary_key=True, autoincrement=True, unique=True)
+    date = Column(types.DateTime, default=datetime.now, primary_key=True)
+    # TODO: add returning visits, new visits
```

Comment:
TODO here relevant?

Reply:
no not at the moment we don't need a separation between new and returning  visitors